addons/project/models/project.py

```diff
@@ -591,16 +585,17 @@ def _read_group_stage_ids(self, stages, domain, order):
         index=True,
         copy=False,
         readonly=True)
-    project_id = fields.Many2one('project.project', string='Project',
-        compute='_compute_project_id', store=True, readonly=False,
+    project_id = fields.Many2one('project.project', string='Project', store=True, readonly=False,
         index=True, tracking=True, check_company=True, change_default=True)
+    effective_project_id = fields.Many2one('project.project',
+        compute='_compute_effective_project_id', store=True, index=True, tracking=True, check_company=True, change_default=True)
```

Comment:
Why splitting the concept of project ? As you need to replace all the occurrences of ``record.project_id`` into ``record.effective_project_id``, this change seems quite louche.

Required a migration script too as the field is stored.

Reply:
I confirm this is louche ! :) Motivation and reasons of this change are written both in commit message and pull request message but live conversations are still mandatory to discuss this choice. 

Upgrade query can be seen here : https://github.com/odoo/upgrade/blob/f2a4c39d5c47c167a3fc064206fafac5a020eba7/migrations/project/saas~14.1.1.1/pre-migrate.py#L11-L18